src/server/scripts/Battlegrounds/BattleForGilneas/battleground_battle_for_gilneas.cpp

```diff
@@ -15,18 +15,299 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "Battleground.h"
+#include "BattlegroundPackets.h"
 #include "BattlegroundScript.h"
 #include "ScriptMgr.h"
 
+namespace BattleForGilneas
+{
+    namespace GameObjects
+    {
+        static constexpr uint32 CapturePointLighthouse = 228050;
```

Comment:
why not going with enums instead? Since you already have a namespace for Battle for Gilneas you can also just go with enum GameObjects

Reply:
With this we can get warnings about unused variables, don't think we can with enums

Reply:
Turning enums into variables only for extra warnings doesn't sound like something that's worth the trouble, considering that you're basically allocating memory for compile-checks only, which IMO should be either handled via static asserts or runtime asserts, informing the user that something is missing. Like for example you should define a maximum number of expected capture points and throw and assert if the setup did not register all of them.

Reply:
Notice they are flagged as constexp. Their usages are literally replaced with its values at compile time, theres no memory disadvantage at runtime